archive_viewer/mixins/traces_table.py

```diff
@@ -177,55 +185,91 @@ def import_csv(self) -> None:
+
+        # Define the list of calculator buttons.
+        # It's a bunch of preset buttons, but users can type other functions under math.
+        buttons = ["7",       "8",     "9",      "+",     "(",      ")",
+                   "4",       "5",     "6",      "-",    "^2", "sqrt()",
+                   "1",       "2",     "3",      "*",   "^-1",  "ln()",
+                   "0",       "e",    "pi",      "/", "sin()", "asin()",
+                   ".",   "abs()", "min()",      "^", "cos()", "acos()",
+                   "PV",  "Clear", "max()", "mean()", "tan()", "atan()"]
 
         # Create the calculator buttons and connect them to the input field
         grid_layout = QGridLayout()
         for i, button_text in enumerate(buttons):
             button = QPushButton(button_text, self)
-            row = i // 4
-            col = i % 4
+            row = i // 6
+            col = i % 6
             grid_layout.addWidget(button, row, col)
-
             # Connect the button clicked signal to the appropriate action
+            # PV currently does nothing, this is a remnant
+            # From when we would have the pv_list open in a new window
             if button_text == "PV":
-                button.clicked.connect(lambda _: self.field.insert("PV"))
+                # TODO: Either give a function for this button or replace it
+                pass
             elif button_text == "Clear":
                 button.clicked.connect(lambda _: self.field.clear())
-            elif button_text == "=":
-                button.clicked.connect(self.evaluate_formula)
             else:
                 button.clicked.connect(lambda _, text=button_text: self.field.insert(text))
-
         layout.addLayout(grid_layout)
 
-        # Add an input field for PV name
-        self.pv_name_input = QLineEdit(self)
-        self.pv_name_input.setPlaceholderText("Enter PV name")
-        layout.addWidget(self.pv_name_input)
-
         # Add an "OK" button to accept the formula and close the dialog
         ok_button = QPushButton("OK", self)
         ok_button.clicked.connect(self.accept_formula)
         layout.addWidget(ok_button)
+    def keyPressEvent(self, e: QKeyEvent) -> None:
+        # Special key press tracker, just so that if enter or return is pressed the formula dialog attempts to submit the formula
+        if e.key() == Qt.Key_Return or e.key() == Qt.Key_Enter:
+            self.accept_formula()
+        return super().keyPressEvent(e)
+    def exec_(self):
+        # When the formula dialog is opened (every time) we need to
+        # update it with the latest information on the curve model and
+        # also populate the text box with the pre-existing formula (if it already was there)
+        self.index = self.parent().selected_index
+        self.pv_list.setRowHidden(len(self.curveModel._row_names) - 1, True)
+        for i in range(self.curveModel.rowCount() - 1):
+            self.pv_list.setRowHidden(i, False)
+        if self.index:
```

Comment:
I believe that exec for all tables is called, which means that this exec is called even though no formula dialogue is being made. At init, no row is selected, so index = None. I don't remember having the issue, but knowing the way I code, this is likely because of that one edge case when exec is called *without* user input

Reply:
Ok, I just tested it and found it unnecessary so I will remove